cylc/flow/config.py

```diff
@@ -178,15 +180,15 @@ def __init__(
             'SequenceBase', Set[Tuple[str, str, bool, bool]]
         ] = {}
         self.taskdefs: Dict[str, TaskDef] = {}
-        self.initial_point: Optional['PointBase'] = None
-        self.start_point: Optional['PointBase'] = None
-        self.final_point: Optional['PointBase'] = None
         self.clock_offsets = {}
         self.expiration_offsets = {}
         self.ext_triggers = {}  # Old external triggers (client/server)
         self.xtrigger_mgr = xtrigger_mgr
         self.workflow_polling_tasks = {}  # type: ignore # TODO figure out type
 
+        self.initial_point: 'PointBase'
+        self.start_point: 'PointBase'
```

Comment:
I didn't explain; it's because they should always be set by https://github.com/cylc/cylc-flow/blob/135ab1aceac04cfa9dfa40b631dbd9c3de3bb459/cylc/flow/config.py#L599 and https://github.com/cylc/cylc-flow/blob/135ab1aceac04cfa9dfa40b631dbd9c3de3bb459/cylc/flow/config.py#L647

(`start_point` takes on value of `initial_point` if not specified)

This stops mypy errors complaining that `start_point` or `initial_point` may be `None` when trying to use `PointBase` methods later.

Only problem is, if they somehow don't end up getting set, we'd end up with `AttributeError`s, which won't get picked up by mypy. But in that case, we'd still get `TypeError`s if they were set to `None` in `__init__` like before, I guess. Not sure what the best practice is, but at least this silences mypy.